addons/addon-raas-s3-copy/packages/s3-synchronizer/src/s3-synchronizer.go

```diff
@@ -19,10 +19,13 @@ func main() {
 
 	sess := makeSession(profile, region)
 
-	mainImpl(sess, debug, recurringDownloads, stopRecurringDownloadsAfter, downloadInterval, concurrency, defaultS3Mounts, destinationBase)
+	// Passing stopUploadWatchersAfter as -1 to let file watchers continue indefinitely if mount is writeable
+	stopUploadWatchersAfter := -1
```

Comment:
Can you remind me the use of it in read-only use case as it seems like we are disabling it for read/write? Is it for unit testing mostly?

Reply:
stopUploadWatchersAfter is for stopping the writer thread. This is only for unit tests and is NOT exposed via the program arguments.